platform/security/servlet/security-servlet-whoami/src/test/groovy/org/codice/ddf/security/servlet/whoami/SubjectSpecification.groovy

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.servlet.whoami
+
+import ddf.security.Subject
+import ddf.security.assertion.SecurityAssertion
+import ddf.security.principal.GuestPrincipal
+import org.apache.shiro.subject.PrincipalCollection
+import org.opensaml.core.xml.schema.XSString
+import org.opensaml.saml.saml2.core.Attribute
+import org.opensaml.saml.saml2.core.AttributeStatement
+import org.opensaml.saml.saml2.core.AuthnStatement
+import spock.lang.Specification
+
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.util.stream.Collectors
+
+class SubjectSpecification extends Specification {
+
+    Date notOnOrAfter = Date.from(LocalDateTime.now()
+            .plusDays(7)
+            .atZone(ZoneId.systemDefault()).toInstant())
+
+    def mockSubject() {
+        def subject = Mock(Subject.class)
+        def pc = Mock(PrincipalCollection.class)
+        def assertion = Mock(SecurityAssertion.class)
+        def principal = new GuestPrincipal("127.0.0.1")
+        def attrStatement = Mock(AttributeStatement.class)
+        def authnStatement = Mock(AuthnStatement.class)
+
+        def attributes = ['http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress':
+                                  ['guest@localhost']]
+
+        def attrs = attributes.entrySet()
```

Comment:
✏️ Totally correct, but weird to see Java 8 lambdas used in Groovy code. :)

Alternatively:
```groovy
        def attrs = attributes.collect {
            mockAttribute(it)
        }
```